docs/Upgrade.md

```diff
@@ -1816,18 +1816,55 @@ The form components don't handle redirection anymore, as redirections are now ha
 
 To upgrade, replace all occurrences of `<FormWithRedirect>` with  `<Form>`.
 
+Also, the `render` prop has been removed, and the form content is now passed as child of the `<Form>` element.
+
 ```diff
 - import { FormWithRedirect } from 'react-admin';
 + import { Form } from 'react-admin';
 
 export const MyForm = () => (
 -    <FormWithRedirect
-+    <Form
-        render={() => ...}
-    />
+-        render={() => ...}
+-    />
++    <Form>
++        ...
```

Comment:
this isn't explicit enough. You should replace the `...` in the diff by a dummy component to make the change clearer